src/dump.c

```diff
@@ -841,6 +841,7 @@ static void jl_serialize_value_(ios_t *s, jl_value_t *v)
         write_int32(s, m->line);
         jl_serialize_value(s, (jl_value_t*)m->sig);
         jl_serialize_value(s, (jl_value_t*)m->tvars);
+        jl_serialize_value(s, (jl_value_t*)m->ambig);
```

Comment:
I'm also a little uncertain about whether one should add this, or whether there's a scan for ambiguities upon deserialization.